container-engine-lib/lib/stream_writer/stream_writer.go

```diff
@@ -0,0 +1,35 @@
+package stream_writer
+
+import (
+	"io"
+	"sync"
+	"time"
+)
+
+// Enables extracting output that is being written to an io.Writer via a channel
+type StreamWriter struct {
+	underlying    io.Writer
+	mutex         *sync.Mutex
+	outputChannel chan string
+}
+
+func NewStreamWriter(underlying io.Writer, outputChannel chan string) *StreamWriter {
+	return &StreamWriter{
+		underlying:    underlying,
+		mutex:         &sync.Mutex{},
+		outputChannel: outputChannel,
+	}
+}
+
+func (writer *StreamWriter) Write(p []byte) (n int, err error) {
+	writer.mutex.Lock()
+	defer writer.mutex.Unlock()
```

Comment:
So I was actually following the ex of[ `ConcurrentWriter` ](https://github.com/kurtosis-tech/kurtosis/blob/44d3b16528a8523f3c20842e373b3b51458fb267/container-engine-lib/lib/concurrent_writer/concurrent_writer.go#L8)here - `bytes.Buffer` is not [thread safe](https://stackoverflow.com/questions/19646717/is-the-go-bytes-buffer-thread-safe) - which is why the lock exists. This only applies if `underlying` is a `bytes.Buffer` tho so maybe I should change the struct to take in a `bytes.Buffer` since it seems that was the intended use? wdyt?